pkg/tsdb/intervalv2/intervalv2.go

```diff
@@ -82,31 +82,40 @@ func (ic *intervalCalculator) CalculateSafeInterval(timerange backend.TimeRange,
 // queryIntervalMS is a pre-calculated numeric representation of the query interval in milliseconds.
 func GetIntervalFrom(dsInterval, queryInterval string, queryIntervalMS int64, defaultInterval time.Duration) (time.Duration, error) {
 	// Apparently we are setting default value of queryInterval to 0s now
-	if queryInterval == "0s" {
-		queryInterval = ""
+	interval := queryInterval
```

Comment:
This is just personal preference/clean up to use the same variable to store interval string.